docs/program-setup/bare_metal/mbed2_porting.md

```diff
@@ -0,0 +1,109 @@
+# Porting an Mbed 2 target to Mbed OS 6 bare metal
+
+The Mbed OS bare metal profile removes the RTOS and provides fewer features compared to Mbed OS 6. This profile offers the same functionality as Mbed 2 and allows targets to access features that we've added to Mbed OS 5 and 6. This document describes how to configure an Mbed 2 target to run in Mbed OS 6 bare metal and how to validate the port.
+
+## Configuring your target
+
+You will use the blinky bare metal Mbed OS example and make it run on your target. The configuration will be validated upon successful run of the example.
+
+The main steps are:
+1. Clone [mbed-os-example-blinky-baremetal](https://github.com/ARMmbed/mbed-os-example-blinky-baremetal).
+1. Edit `targets.json`. In this step, you will configure your target to support Mbed OS 6 and add bare metal configuration parameters.
+1. Compile and run mbed-os-example-blinky-baremetal. In this step, you will validate the configuration and troubleshoot any issue.
+
+### Clone mbed-os-example-blinky-baremetal
+
+Run the following commands:
+1. `git clone https://github.com/ARMmbed/mbed-os-example-blinky-baremetal`
+1. `cd mbed-os-example-blinky-baremetal`
+
+### Edit targets.json
+
+Find your target in `mbed-os/targets/targets.json` and update its configuration as described below.
+
+Configure your target to support Mbed OS 6:
+- Remove the `release_versions` property as it is no longer necessary to specify the list of major versions of Mbed OS that the target supports.
+```
+"release_versions": ["2"]
+```
+
+- Add the `supported_application_profiles` property to indicate that the application profile supported by this target is bare metal. This property is currently ignored by the build tools as it is not yet supported.
```

Comment:
Why is this attribute needed? How will it be used by the tools? Will the target that can work with an RTOS and bare metal also have it? If so, what will be the value of the attribute for those targets?

Reply:
I would drop the second part of this paragraph `This property is currently ignored by the build tools as it is not yet supported.` it doesn't add anything. And it raises questions if it's not supported why do we ask people to add it.